sigstore/_verify.py

```diff
@@ -244,7 +244,11 @@ def verify(
         )
         if entry is None:
             return VerificationFailure(
-                reason="Rekor has no entry for these verification materials"
+                reason=(
+                    "Rekor has no entry for these verification materials; "
+                    "this suggests that the signing certificate was not published "
+                    "to the log at the time the signature was created"
```

Comment:
I'm curious to hear feedback on this -- Rekor missing the certificate is (IMO) the most likely reason for failure here, although it could also happen because the user searches the wrong tlog (e.g. staging instead of production) or if they signed multiple artifacts using the certificate but only uploaded signatures for a subset of them.

Reply:
Yeah, agreed. I think I'm going to do what we did in `pip-audit` here and specialize `VerificationFailure` a bit more, so that we can push some of this error handling into the CLI. That in turn will allow us to include the filename and tlog information, since we have more context there.

Reply:
[967a587](https://github.com/sigstore/sigstore-python/pull/241/commits/967a58751e2e1ac88b6ab586f2ed4944da18ddf4) addresses most of this -- the artifact filename is emitted as part of the top-level failure message, and we now emit additional context (like the artifact hash and signature) that should help a user do a manual tlog lookup.